_applications/equoid.adoc

```diff
@@ -0,0 +1,206 @@
+
+[[architecture]]
+== Microservice Architecture
+
+Equoid is made up of a number of components giving users a means for visualizing the top-k elements in a stream during a specified temporal window. This is made possible through a hosted web UI and backend REST interface for parameter updates. 
+
+The services of Equoid are: 
+
+- A data publisher
+- A message broker
+- A data handler
+- A caching layer 
+- A web UI
+- A REST API for adding items to the set of possibilities the data publisher chooses from 
+
+The following diagram shows the overall architecture of Equoid.
+
+pass:[<img src="/assets/equoid/Equoid-Architecture.png" alt="Alt text" class="img-responsive arch" width="800px">]
+
+Equoid utilizes a caching layer for storing the top-k values. The data publisher transmits records to Artemis, the data handler retrieves mesages from the Artemis broker, parses or converts them per user specification, then adds the information to the extant top-k object. It's worth noting that multiple instances of the data publisher and data handler can co-exist, which would be akin to a production system. 
+
+== Software Components
+
+We chose to use Artemis for the message broker, with the expectation it would be scalable in terms of number of streams as well as volume of traffic. In the same vein, we utilized Spark for our microservice requiring scalability, namely the data handler. Since we wished to take advantage of extant Spark libraries, and the CMSketch() object is only (as of this writing) available in Java and Scala, we chose to implement the handler in Scala. In the interest of uniformity the sample data publisher is also written in Scala. Furthermore, the AMQP Spark connector used by the data handler is one developed by Paolo Patierno at Red Hat, with more information https://radanalytics.io/examples/amqpstreaming[this tutorial] and associated https://github.com/redhat-iot/amqp-spark-demo[upstream project].
```

Comment:
`CMSketch()`

the brackets

> ..In the interest of uniformity the sample data publisher..

My English sucks, but shouldn't there be a comma between "uniformity" and "the"?

> .. is one developed by Paolo Patierno..

My English sucks, but `is one` -?> `is the one`

Reply:
Revised accordingly, thanks for the suggestions